tez-api/src/test/java/org/apache/tez/client/TestTezClient.java

```diff
@@ -447,6 +457,94 @@ public TezClientForTest testTezClient(boolean isSession, boolean shouldStop, Str
     return client;
   }
 
+  public void testTezClientReconnect(boolean isSession) throws Exception {
+    //Setup 1
+    Map<String, LocalResource> lrs = Maps.newHashMap();
+    String lrName1 = "LR1";
+    lrs.put(lrName1, LocalResource.newInstance(URL.newInstance("file", "localhost", 0, "/test"),
+            LocalResourceType.FILE, LocalResourceVisibility.PUBLIC, 1, 1));
+
+    //Client 1
+    TezClientForTest client = configureAndCreateTezClient(lrs, isSession, null);
+
+    //Submission Context 1
+    ArgumentCaptor<ApplicationSubmissionContext> captor = ArgumentCaptor.forClass(ApplicationSubmissionContext.class);
+    when(client.mockYarnClient.getApplicationReport(client.mockAppId).getYarnApplicationState())
+            .thenReturn(YarnApplicationState.RUNNING);
+
+    //Client 1 start
+    client.start();
```

Comment:
we are starting client and all, but no closing, is there a proper cleanup somewhere, so it doesn't leak resources or so, can you just double check once

Reply:
yeah, dagClient can and should be closed
I cannot see separate close method in TezClient